dns/src/adapter.ts

```diff
@@ -0,0 +1,42 @@
+import { Execute } from '@chainlink/types'
+import { Requester, Validator } from '@chainlink/external-adapter'
+
+const customParams = {
+  name: true,
+  type: true,
+  do: false,
+  cd: false,
+}
+
+const baseUrl = 'https://cloudflare-dns.com/dns-query'
+
+export const execute: Execute = async (input) => {
```

Comment:
The important thing with this data provider adapter is to add appropriate types and then map to/from those types for specific underlying data providers.

We are missing an input type, lets call it `DNSDataRequest`, and also an output type `DNSDataResponse`.

Then when we define those we can have N DNS data providers answering to our `DNSDataRequest` with `DNSDataResponse`. If the underlying API uses different types, we need to map from/to those types.

Then we'll have N DNS data providers that have the same API and we can easily build composite adapters from it, which are nothing more then simple data processing pipelines that use underlying data providers.

Reply:
The DNS adapter factory is the base package where we can define all those types, configuration and validation. The the specific implementations would then just map from `DNSDataRequest` to the actual API request, and back from the response to the `DNSDataResponse`.